src/main/java/org/spongepowered/api/effect/sound/SoundEffect.java

```diff
@@ -0,0 +1,291 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.effect.sound;
+
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.data.persistence.DataBuilder;
+import org.spongepowered.api.data.persistence.DataSerializable;
+import org.spongepowered.api.effect.PlayableEffect;
+import org.spongepowered.api.util.ResettableBuilder;
+
+import java.util.function.Supplier;
+
+/**
+ * Represents a sound effect that can be played.
+ */
+public interface SoundEffect extends PlayableEffect, DataSerializable {
+
+    /**
+     * Constructs a new {@link SoundEffect}.
+     *
+     * @param type The type
+     * @param category The category
+     * @param volume The volume
+     * @param pitch The pitch
+     * @return The sound effect
+     */
+    static SoundEffect of(SoundType type, SoundCategory category, double volume, double pitch) {
+        return builder().type(type).category(category).volume(volume).pitch(pitch).build();
+    }
```

Comment:
Not sure how I feel about `of`s with more than 3 parameters. I know this is just mirroring the old `playSounds` methods, but doesn't that defeat the purpose of using builders in the first place?

Reply:
Yeah, I was also thinking about that. What about just the methods with the type, volume and pitch?

Reply:
All the other values' defaults seem to follow principle of least surprise, so yea seems good to me. 👍

Reply:
Any `of` with more than a few is gonna be too much.